requirements.txt

```diff
@@ -2,4 +2,5 @@ Mako
 PyYAML
 six
 pbr
-defusedxml
+# For OpenNebula provider
+# defusedxml
```

Comment:
OK, this works: https://github.com/gwarf/cloud-info-provider/commit/20471fd04d12aa01114e9a81e734ac299cdfa262  and then you can use
```sh
pip install .'[opennebula]'                                                                                  130 ⏎  +10075 18:41 ❰─┘ 
Processing /home/baptiste/repos/cloud-info-provider                                                                                                                        
Collecting Mako (from cloud-info-provider==0.8.3)                                                                                                                          
Collecting PyYAML (from cloud-info-provider==0.8.3)                                                                                                                        
Collecting six (from cloud-info-provider==0.8.3)                                                                                                                           
  Using cached six-1.10.0-py2.py3-none-any.whl                                                                                                                             
Collecting pbr (from cloud-info-provider==0.8.3)                                     
  Using cached pbr-3.1.1-py2.py3-none-any.whl                                                                                                                              
Collecting defusedxml (from cloud-info-provider==0.8.3)                              
  Using cached defusedxml-0.5.0-py2.py3-none-any.whl                                                                                                                       
Collecting MarkupSafe>=0.9.2 (from Mako->cloud-info-provider==0.8.3)                                                                                                 
Building wheels for collected packages: cloud-info-provider                                                                                                                
  Running setup.py bdist_wheel for cloud-info-provider ... done                                                   
  Stored in directory: /home/baptiste/.cache/pip/wheels/e3/f2/28/1cdc72c1d369f1eaedc02931b47e67d4dbb2f93baf63f8500c                                                        
Successfully built cloud-info-provider                                                                                                                                     
Installing collected packages: MarkupSafe, Mako, PyYAML, six, pbr, defusedxml, cloud-info-provider                                                                         
Successfully installed Mako-1.0.6 MarkupSafe-1.0 PyYAML-3.12 cloud-info-provider-0.8.3 defusedxml-0.5.0 pbr-3.1.1 six-1.10.0 
```
or
```sh
pip install .'[openstack]'                                                                                         +10078 1[68/1717]
Processing /home/baptiste/repos/cloud-info-provider                                                                                                                        
Collecting Mako (from cloud-info-provider==0.8.3)                                                                                                                          
Collecting PyYAML (from cloud-info-provider==0.8.3)
Collecting six (from cloud-info-provider==0.8.3)
  Using cached six-1.10.0-py2.py3-none-any.whl
Collecting pbr (from cloud-info-provider==0.8.3)
  Using cached pbr-3.1.1-py2.py3-none-any.whl
Collecting python-novaclient (from cloud-info-provider==0.8.3)
  Using cached python_novaclient-9.0.1-py2.py3-none-any.whl
Collecting MarkupSafe>=0.9.2 (from Mako->cloud-info-provider==0.8.3)
Collecting keystoneauth1>=2.21.0 (from python-novaclient->cloud-info-provider==0.8.3)
  Using cached keystoneauth1-2.21.0-py2.py3-none-any.whl
Collecting simplejson>=2.2.0 (from python-novaclient->cloud-info-provider==0.8.3)
Collecting Babel!=2.4.0,>=2.3.4 (from python-novaclient->cloud-info-provider==0.8.3)
  Using cached Babel-2.3.4-py2.py3-none-any.whl
Collecting iso8601>=0.1.11 (from python-novaclient->cloud-info-provider==0.8.3)
  Using cached iso8601-0.1.11-py2.py3-none-any.whl
Collecting oslo.utils>=3.20.0 (from python-novaclient->cloud-info-provider==0.8.3)
  Using cached oslo.utils-3.26.0-py2.py3-none-any.whl
Collecting PrettyTable<0.8,>=0.7.1 (from python-novaclient->cloud-info-provider==0.8.3)
Collecting oslo.serialization>=1.10.0 (from python-novaclient->cloud-info-provider==0.8.3)
  Using cached oslo.serialization-2.19.0-py2.py3-none-any.whl
Collecting oslo.i18n!=3.15.2,>=2.1.0 (from python-novaclient->cloud-info-provider==0.8.3)
  Using cached oslo.i18n-3.15.3-py2.py3-none-any.whl
Collecting requests!=2.12.2,!=2.13.0,>=2.10.0 (from keystoneauth1>=2.21.0->python-novaclient->cloud-info-provider==0.8.3)
  Using cached requests-2.18.1-py2.py3-none-any.whl
Collecting positional>=1.1.1 (from keystoneauth1>=2.21.0->python-novaclient->cloud-info-provider==0.8.3)
Collecting stevedore>=1.20.0 (from keystoneauth1>=2.21.0->python-novaclient->cloud-info-provider==0.8.3)
  Using cached stevedore-1.23.0-py2.py3-none-any.whl
Collecting pytz>=0a (from Babel!=2.4.0,>=2.3.4->python-novaclient->cloud-info-provider==0.8.3)
  Using cached pytz-2017.2-py2.py3-none-any.whl
Collecting funcsigs>=0.4; python_version == "2.7" or python_version == "2.6" (from oslo.utils>=3.20.0->python-novaclient->cloud-info-provider==0.8.3)
  Using cached funcsigs-1.0.2-py2.py3-none-any.whl
Collecting netaddr!=0.7.16,>=0.7.13 (from oslo.utils>=3.20.0->python-novaclient->cloud-info-provider==0.8.3)
  Using cached netaddr-0.7.19-py2.py3-none-any.whl
Collecting netifaces>=0.10.4 (from oslo.utils>=3.20.0->python-novaclient->cloud-info-provider==0.8.3)
Collecting pyparsing>=2.1.0 (from oslo.utils>=3.20.0->python-novaclient->cloud-info-provider==0.8.3)
  Using cached pyparsing-2.2.0-py2.py3-none-any.whl
Collecting monotonic>=0.6 (from oslo.utils>=3.20.0->python-novaclient->cloud-info-provider==0.8.3)
  Using cached monotonic-1.3-py2.py3-none-any.whl
Collecting debtcollector>=1.2.0 (from oslo.utils>=3.20.0->python-novaclient->cloud-info-provider==0.8.3)
  Using cached debtcollector-1.15.0-py2.py3-none-any.whl
Collecting msgpack-python>=0.4.0 (from oslo.serialization>=1.10.0->python-novaclient->cloud-info-provider==0.8.3)
Collecting idna<2.6,>=2.5 (from requests!=2.12.2,!=2.13.0,>=2.10.0->keystoneauth1>=2.21.0->python-novaclient->cloud-info-provider==0.8.3)
  Using cached idna-2.5-py2.py3-none-any.whl
Collecting urllib3<1.22,>=1.21.1 (from requests!=2.12.2,!=2.13.0,>=2.10.0->keystoneauth1>=2.21.0->python-novaclient->cloud-info-provider==0.8.3)
  Using cached urllib3-1.21.1-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests!=2.12.2,!=2.13.0,>=2.10.0->keystoneauth1>=2.21.0->python-novaclient->cloud-info-provider==0.8.3)
  Using cached chardet-3.0.4-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests!=2.12.2,!=2.13.0,>=2.10.0->keystoneauth1>=2.21.0->python-novaclient->cloud-info-provider==0.8.3)
  Using cached certifi-2017.4.17-py2.py3-none-any.whl
Collecting wrapt (from positional>=1.1.1->keystoneauth1>=2.21.0->python-novaclient->cloud-info-provider==0.8.3)
Building wheels for collected packages: cloud-info-provider
  Running setup.py bdist_wheel for cloud-info-provider ... done
  Stored in directory: /home/baptiste/.cache/pip/wheels/e3/f2/28/1cdc72c1d369f1eaedc02931b47e67d4dbb2f93baf63f8500c
Successfully built cloud-info-provider
Installing collected packages: MarkupSafe, Mako, PyYAML, six, pbr, idna, urllib3, chardet, certifi, requests, iso8601, wrapt, positional, stevedore, keystoneauth1, simplej
son, pytz, Babel, funcsigs, netaddr, netifaces, pyparsing, monotonic, debtcollector, oslo.i18n, oslo.utils, PrettyTable, msgpack-python, oslo.serialization, python-novacli
ent, cloud-info-provider
Successfully installed Babel-2.3.4 Mako-1.0.6 MarkupSafe-1.0 PrettyTable-0.7.2 PyYAML-3.12 certifi-2017.4.17 chardet-3.0.4 cloud-info-provider-0.8.3 debtcollector-1.15.0 f
uncsigs-1.0.2 idna-2.5 iso8601-0.1.11 keystoneauth1-2.21.0 monotonic-1.3 msgpack-python-0.4.8 netaddr-0.7.19 netifaces-0.10.6 oslo.i18n-3.15.3 oslo.serialization-2.19.0 os
lo.utils-3.26.0 pbr-3.1.1 positional-1.1.1 pyparsing-2.2.0 python-novaclient-9.0.1 pytz-2017.2 requests-2.18.1 simplejson-3.11.1 six-1.10.0 stevedore-1.23.0 urllib3-1.21.1
 wrapt-1.10.10
```
I can submit a PR for this.